cli/azd/internal/repository/infra_confirm.go

```diff
@@ -107,6 +108,11 @@ func (i *Initializer) infraSpecFromDetect(
 			if svc.Language == appdetect.Java {
 				serviceSpec.Port = 8080
 			}
+		} else {
+			ports := i.detectPortInDockerfile(svc.Docker.Path)
+			if len(ports) == 1 {
+				serviceSpec.Port = ports[0]
+			}
```

Comment:
@weikanglim 

In the issue description of #4443, you mentioned that

> 3. If EXPOSE is not found, default the port for customer.

I don't know what the `default the port` is, so I just keep original behavior: Not set the port, `ask the customer`.

Reply:
Asking the port is indeed the right behavior! I updated the issue

Reply:
@rujche I also updated the multiple ports requirement:

4. If more than one port is exposed, prompt for a selection (`console.Select`) with the multiple ports available, and an "Other" selection to provide the port number.
   
Multiple ports example:
```bash
Select a port number:
- 5050 
- 8080
- Other
```